### DISCUSSION TITLE
spotify for the music bot

### DISCUSSION THREAD

**Discussion by SaboiFromHanoi** [jagrosh/MusicBot#797] [Ideas]
any chance of the music bot being able to support Spotify.

**Comment by jagrosh** [comment]
It is abundantly clear from [this page](https://github.com/jagrosh/MusicBot/wiki/Things-That-Won%27t-Be-Added#spotify) that spotify support cannot be added.

**Reply by Alppuccino** [reply]
Hi, I understand that playing from spotify isn't possible, but I'd personally be more than happy with getting the closest match from soundcloud/youtube, as Rythm used to do this and we basically never had a problem with that. Reason why I ask is because just doing !p <spotify playlist link> is super easy and means that those without premium can listen along on the bot. Just wondering if this would be possible. Cheers.

**Reply by MichailiK** [reply]
![image](https://user-images.githubusercontent.com/39029839/133496313-018fa677-a7b8-45e5-9543-f3749393d9e7.png)

**Comment by GravelX** [comment]
Not gonna lie this is the only feature making me change to another bot. Not being able to play spotify playlists makes it so you have to manually type /p 20 times in a row to build a playlist... Adding an API key would be optionnal so the user-friendly argument doesn't work. If streaming from spotify isn't allowed... how are hundreds of other bots doing it lol? "Discord already has native support for syncing up Spotify playlists"... that requires that every person in the channel clicks on someone's profile and click a button. Not gonna happen for a quick song or if you want to switch from youtube song to a spotify song. Finding the closest match on Youtube would be more than enough.

**Reply by MichailiK** [reply]
> Not being able to play spotify playlists makes it so you have to manually type /p 20 times in a row to build a playlist...

There seem to be [a few tools](https://www.google.com/search?q=spotify%20playlist%20to%20youtube%20playlist) which allow you to convert a Spotify playlist to a YouTube playlist.

>  Adding an API key would be optionnal so the user-friendly argument doesn't work.

JMusicBot is intended as a bot that is easy to set up & use. API keys, or any other form of authentication token, except Discord tokens, makes JMusicBot more complicated & confusing to use.

> If streaming from spotify isn't allowed... how are hundreds of other bots doing it lol?

From [Things That Won't Be Added # Spotify](https://github.com/jagrosh/MusicBot/wiki/Things-That-Won%27t-Be-Added#spotify)
> 2. Streaming from Spotify isn't allowed, **so the bot would have to "guess" and just play the closest match from YouTube, which is unreliable at best**.

> Finding the closest match on Youtube would be more than enough.

Since finding the closest match on YouTube is considered to be unreliable (at best), it would likely degrade user experience. Thus, this feature will not be implemented.

**Reply by GravelX** [reply]
Yeah I know, was just letting you guys know you're losing users because of that weird decision. Good luck with your project!

**Reply by ghost** [reply]
This guy has a point, I see no reason why you can't add the feature as an optional extra? Maybe an extension? Plus Spotify dev/api keys take a single minute to sign up for and are pretty straight forward.

**Reply by YJSoft** [reply]
Reasons are not needed to decline suggest. Please note that being open source does not mean every suggestions should accepted.

**Reply by Diblox** [reply]
> This guy has a point, I see no reason why you can't add the feature as an optional extra? Maybe an extension? Plus Spotify dev/api keys take a single minute to sign up for and are pretty straight forward.

Spotify doesn't allow streaming from other services. Bots that does that most likely doesn't comply with their TOS.

**Reply by ghost** [reply]
> 
> 
> > This guy has a point, I see no reason why you can't add the feature as an optional extra? Maybe an extension? Plus Spotify dev/api keys take a single minute to sign up for and are pretty straight forward.
> 
> Spotify doesn't allow streaming from other services. Bots that does that most likely doesn't comply with their TOS.

tbf it's also against youtube TOS to do that.
